MBBSEmu/HostProcess/ExportedModules/Majorbbs.cs

```diff
@@ -3375,16 +3349,17 @@ private void rename()
         /// </summary>
         private void rstrin()
         {
-            ChannelDictionary[ChannelNumber].rstrin();
+            var inputLength = Module.Memory.GetWord("INPLEN");
+            var inputPointer = Module.Memory.GetVariablePointer("INPUT");
 
-            var inputMemory = Module.Memory.GetVariablePointer("INPUT");
-            Module.Memory.SetZero(inputMemory, 0xFF);
-            Module.Memory.SetArray(inputMemory.Segment, inputMemory.Offset,
-                ChannelDictionary[ChannelNumber].InputCommand);
+            if (inputLength == 0)
+                return;
 
-#if DEBUG
-            _logger.Info("Restored Input");
-#endif
+            for (var i = inputPointer.Offset; i < inputPointer.Offset + (inputLength - 1); i++)
+            {
+                if (Module.Memory.GetByte(inputPointer.Segment, i) == 0)
+                    Module.Memory.SetByte(inputPointer.Segment, i, 0x20);
```

Comment:
0x20 isn't quite as intuitive as ' ' , for me at least

Reply:
Made it `(byte)' '` ❤️